nvflare/mt/job_def_manager.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import pathlib
+import time
+import uuid
+from typing import Any, Dict, List
+
+from nvflare.apis.job_def_manager_spec import JobDefManagerSpec, RunStatus
+from nvflare.apis.storage import StorageSpec
+from nvflare.apis.study_manager_spec import StudyManagerSpec
+from nvflare.mt.job_def import JobMetaKey, Job, job_from_meta
+
+
+class _JobChecker:
+    def check_job(self, meta: dict) -> bool:
+        pass
+
+
+class _StatusChecker(_JobChecker):
+    def __init__(self, status_to_check):
+        self.result = []
+        self.status_to_check = status_to_check
+
+    def check_job(self, meta: dict):
+        if meta[JobMetaKey.STATUS] == self.status_to_check:
+            self.result.append(meta)
+        return True
+
+
+class _AllJobsChecker(_JobChecker):
+    def __init__(self):
+        self.result = []
+
+    def check_job(self, meta: dict):
+        self.result.append(meta)
+        return True
+
+
+class _ReviewerChecker(_JobChecker):
+    def __init__(self, reviewer_name, study_manager: StudyManagerSpec):
+        self.result = []
+        self.reviewer_name = reviewer_name
+        self.study_manager = study_manager
+
+    def check_job(self, meta: dict):
+        study = self.study_manager.get_study(meta[JobMetaKey.STUDY_NAME])
+        if study and study.reviewers and self.reviewer_name in study.reviewers:
+            # this job requires review from this reviewer
+            # has it been reviewed already?
+            approvals = meta.get(JobMetaKey.APPROVALS)
+            if approvals and self.reviewer_name in approvals:
+                # already reviewed
+                return True
+            else:
+                self.result.append(meta)
+        return True
+
+
+class SimpleJobDefManager(JobDefManagerSpec):
+    def __init__(self, study_manager: StudyManagerSpec, store: StorageSpec, uri_root: str = "jobs"):
+        super().__init__()
+        self.store = store
+        self.uri_root = uri_root
+        self.result_uri_root = "results"
+        self.study_manager = study_manager
+
+    def job_uri(self, jid: str):
+        return "/" + self.uri_root + "/" + jid
```

Comment:
I suggest we remove "/" in the front.
For the second "/", suggest use os.path.join or os.sep